src/Simulation/Simulators/QuantumSimulator/ApplyUncontrolledSWAP.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.Simulators
+{
+    public partial class QuantumSimulator
```

Comment:
Maybe add a link to https://github.com/microsoft/qsharp-runtime/issues/44?

Reply:
That's a great idea! I'll also add a comment on the issue that points to this part of the implementation to make it easier to find for whoever picks up that work.